Gems/Atom/RHI/Code/Source/RHI/CpuProfilerImpl.cpp

```diff
@@ -379,12 +347,12 @@ namespace AZ
                 // Add the cached regions to the map
                 for (auto& cachedTimeRegion : m_cachedTimeRegions)
                 {
-                    const AZStd::string regionName = cachedTimeRegion.m_groupRegionName->m_regionName;
+                    const AZStd::string regionName = cachedTimeRegion.m_groupRegionName.m_regionName;
                     AZStd::vector<CachedTimeRegion>& regionVec = m_cachedTimeRegionMap[regionName];
                     regionVec.push_back(cachedTimeRegion);
                     if (regionVec.size() >= TimeRegionStackSize)
                     {
-                        m_hitSizeLimitMap[cachedTimeRegion.m_groupRegionName->m_regionName] = true;
+                        m_hitSizeLimitMap[cachedTimeRegion.m_groupRegionName.m_regionName] = true;
                     }
```

Comment:
The region name is wasted while it could potentially be reused by replacing line 355 with `m_hitSizeLimitMap[AZStd::move(regionName)] = true;`. However even this is not as efficient as `m_hitSizeLimitMap.insert_or_assign(AZStd::move(regionName), true);`
A more general problem is the fact that line 353 doesn't always goes into the branch so there's also situations where the `regionName` string is copied for no reason other than to look it up in `m_cachedTimeRegionMap`. I'm wondering if both maps actually need to store the string or maybe can also get away with a `AZStd::string_view` to avoid string copies at all.

Reply:
Good point, I'll look into it

Reply:
Great callout, but I'm inclined to say there may not be any harm in deferring "actual" perf improvements of the Atom CPU profiler to a later change (where we can actually "profile the profiler" as they say and do a more comprehensive pass, taking this item into account). If this improvement ends up being easy, or we just opt to move the string, then we might as well include the change in this PR, but beyond that we probably should just make a note of it.

Reply:
For reference, `m_hitSizeLimitMap` was somewhat of a hack to get around the profiler killing performance when we had 100k+ root regions within one frame (see discussion in #1636). If we can restrict profiling regions to only "good" ones (not invoked excessively per frame) or have a less locky profiler implementation this could be removed entirely.

Reply:
@jeremyong-az There's two suggestions, the first one is low hanging fruit that can be picked up this PR. The bigger chance can wait till later for sure, just wanted to raise awareness to it.